### PR TITLE
feat(samples): Add usage examples for the node.js API

### PR DESCRIPTION
**Pull request by Shnatsel**
Opening as draft because this is missing tests. I would appreciate a review of the snippets in the meanwhile, since the GA launch is coming soon.

The samples are modeled after https://github.com/googleapis/python-batch/pull/22, following the Python version closely.

### CODE REVIEW

File: packages/google-cloud-batch/samples/basics/create_with_container_no_mounting.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  job.name = jobName;
+  job.taskGroups = [group];
+  job.allocationPolicy = allocationPolicy;
+  job.labels = {env: 'testing', type: 'container'};
+  // We use Cloud Logging as it's an out option available out of the box
+  job.logs_policy = new batch.LogsPolicy();
+  job.logs_policy.destination = batch.LogsPolicy.Destination.CLOUD_LOGGING;
+
+  // The job's parent is the project and region in which the job will run
+  const parent = 'projects/' + projectId + '/locations/' + region;
```
Comment (FrodoTheTrue) [inline packages/google-cloud-batch/samples/basics/create_with_container_no_mounting.js:85]: would be great to use interpolation here

File: packages/google-cloud-batch/samples/basics/create_with_script_no_mounting.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+function main(projectId, region, jobName) {
```
Comment (FrodoTheTrue) [inline packages/google-cloud-batch/samples/basics/create_with_script_no_mounting.js:17]: Please add jsdoc to the sample, even if it's not included in region tag.
Example:
https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/iam/deny/createDenyPolicy.js#L26

**Review by FrodoTheTrue — COMMENTED** [review]
Overall looks good!
Just couple requests.

Btw, according to the last update, we are going to store all the new samples here: https://github.com/GoogleCloudPlatform/nodejs-docs-samples

There is an example PR how to set up CI for the new folder: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/2763

File: packages/google-cloud-batch/samples/basics/create_with_script_no_mounting.js (hunk 2 above)
Comment (Shnatsel) [reply packages/google-cloud-batch/samples/basics/create_with_script_no_mounting.js:27]: Done

**Comment by Shnatsel** [comment]
Migrating to the monorepo and setting up CI is going to take a while. And Batch launch is coming really soon now. So I'm going to merge the samples into this repo, and migrate them along with numerous other samples en masse at a later date.

Thanks for the review!

**Comment by alexander-fenster** [comment]
@sofisl Should samples actually go here, or to nodejs-docs-samples?

**Comment by Shnatsel** [comment]
it was decided that samples should go in `nodejs-docs-samples`, so closing this in favor of https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/2775

**Pull request closed without merging**